src/cloudWatchLogs/registry/logStreamRegistry.ts

```diff
@@ -178,15 +177,38 @@ export class LogStreamRegistry {
         return (log && log.busy) ?? false
     }
 
-    private setLog(uri: vscode.Uri, stream: CloudWatchLogsData): void {
-        this.activeStreams.set(uri.path, stream)
+    public setLog(uri: vscode.Uri, stream: CloudWatchLogsData): void {
+        this.activeStreams.set(uriToKey(uri), stream)
     }
 
     private getLog(uri: vscode.Uri): CloudWatchLogsData | undefined {
-        return this.activeStreams.get(uri.path)
+        return this.activeStreams.get(uriToKey(uri))
     }
 }
 
+export async function filterLogEventsFromUriComponents(
+    logGroupInfo: CloudWatchLogsGroupInfo,
+    parameters: CloudWatchLogsParameters,
+    nextToken?: string
+): Promise<CloudWatchLogsResponse> {
+    const client = new DefaultCloudWatchLogsClient(logGroupInfo.regionName)
+
+    const response = await client.filterLogEvents({
+        logGroupName: logGroupInfo.groupName,
+        filterPattern: parameters.filterPattern,
+        nextToken,
+        limit: parameters.limit,
+    })
+
+    // Use heuristic of last token as backward token and next token as forward to generalize token form.
+    // Note that this fails if the contents of the calls are changing as they are being made.
+    // However, this fail wouldn't really impact customers.
```

Comment:
Is 'fails' the right word here? Seems more like the results are inconsistent, though it's not strictly a failure.

Reply:
Yeah, I guess I was thinking that it fails to be absolutely true to its function name, but it is still returning the information a customer is presumably looking for, so not really a strict failure.